llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3664,32 +3664,35 @@ static std::optional<DecomposedBitMaskMul> matchBitmaskMul(Value *V) {
   return std::nullopt;
 }
 
-using CombinedBitmaskMul =
-    std::pair<std::optional<DecomposedBitMaskMul>, Value *>;
+struct CombinedBitmaskMul {
+  std::optional<DecomposedBitMaskMul> Decomp = std::nullopt;
```

Comment:
```suggestion
  std::optional<DecomposedBitMaskMul> Decomp;
```

This already default constructs to nullopt